### PR TITLE
Fixes #261 partially

### LINKED ISSUE #261: Add version detect commands and version check regular expressions for all packages in get-generic-sys-util
For detection to properly work we need to add `CM_SYS_UTIL_VERSION_CMD` and `CM_SYS_UTIL_VERSION_RE` like done [here](https://github.com/mlcommons/cm4mlops/blob/mlperf-inference/script/get-generic-sys-util/_cm.json#L641) for `wget`.

### PR DESCRIPTION
**Pull request by Oseltamivir**
- add version detect commands and version check regex
- added description in README

The changes so far has been tested on a ubuntu 22.04 docker image.

Some utilities that are not yet tested:
Packages:
  libmkl-dev
  linux-tools

### CODE REVIEW

**Comment by arjunsuresh** [comment]
Thank you @Oseltamivir for your changes. Can you please tell if you are from an MLCommons member organization? Can you please sign the CLA?

**Comment by Oseltamivir** [comment]
recheck

**Comment by arjunsuresh** [comment]
Thank you @Oseltamivir The CLA test now passes. There is a problem for `md5sum` version detect though as the version output is different on `macos`. Should we change the `re` to `r"\b\d+\.\d+(?:\.\d+)?\b"` or do you have some better idea here? 

```
md5sum --version
Microbrew md5sum/sha1sum/ripemd160sum 0.9.5 (Wed Dec  6 12:48:56 EST 2006)
  Compiled Dec  6 2006 at 17:48:56
Written by Bulent Yilmaz

Copyright (C) 2004,2006 Microbrew Software
```

**Comment by arjunsuresh** [comment]
@Oseltamivir yes, thats correct. Most times the version detection shouldn't change depending on the package manager but we still need to test. Would you like to join the weekly 30 minutes CM sync to discuss on this? It is at GMT 6:05pm Tuesdays. 
 
https://meet.google.com/jtf-crbz-ezz?

**Comment by Oseltamivir** [comment]
I apologise but I dont think I can make it. I'm in GMT+8 which is around 3am... maybe I'll try next week.

**Comment by arjunsuresh** [comment]
No worries @Oseltamivir Let me know a good time and we can sync separately. Usually GMT 9-12 works best for me.

**Comment by Oseltamivir** [comment]
I think next week's CM sync should be fine. I just need to plan ahead. Sleeping nowadays at 3am anyway ¯\_(ツ)_/¯

**Comment by arjunsuresh** [comment]
No worries. Good night! :)

**Comment by arjunsuresh** [comment]
@Oseltamivir Your PR was a big motivation for us to add the `cm test automation` which we now use to test any CM script. For `get-generic-sys-util` the tests are run for all variations on `ubuntu-20.04`, `ubuntu-22.04`, `ubuntu24.04` and `rhel9`. You can see the results [here](https://github.com/mlcommons/cm4mlops/actions/runs/11612866202/job/32337303568?pr=426)

**Comment by Oseltamivir** [comment]
@arjunsuresh Awesome! Also, if you celebrate it, Happy Diwali!

Let me know if there are any regexes/tasks you need me to do regarding this aspect. Meanwhile, I will try to resolve issues and see if i can contribute to this repo and mlcommons/inference - my current focus is attempting to make the default reference implementation use all GPUs on a single node.

**Comment by arjunsuresh** [comment]
Thank you @Oseltamivir It'll be great if this PR is fixed and merged as it adds a very useful feature. 

"reference implementation use all GPUs on a single node"

This is very useful as and some of the reference implementations like for llama2-70b already use all available GPUs on a node. But most of them don't.

**Comment by Oseltamivir** [comment]
Sure. No prob. I'll try to slowly go thru the failing tests

**Comment by Oseltamivir** [comment]
Got very frustrated over a specific library

`pstree --version` prints to `stderr` instead of `stdout`!!!! Spent the last 2 days ripping my hair out over why the regex isn't working. 

Added a specific clause in the `.sh` that detects if command is `pstree --version` which will write to `tmp-ver.out` from `stderr instead of `stdout`. It is a messy way but I think it is sufficient for now unless other libraries write to `stderr`.

Got a feeling the errors for `bzcat --version` in the build tests are also caused by printing to `stderr`...

I will move onto check the other regexes soon. Currently wondering why `dpkg -l | grep libnuma-dev` failed on ResNet50... I think it might not be installed... @arjunsuresh Should we run the install command in `_cm.json` if `tmp-ver.out` is empty?

**Comment by arjunsuresh** [comment]
Oh. That's an interesting find. Thank you for digging into it.

I think it's better to set the below variable in `_cm.json` for the packages that needs error stream so that the bash script becomes cleaner. 


```
env['CM_SYS_UTIL_VERSION_CMD_USE_ERROR_STREAM'] = yes
```
If `tmp-ver.out` is empty, but no error happens, I think we'll need to handle case by case. One option can be to say "version-undetected" but for this we need to make sure that the installation is successful.

**Comment by Oseltamivir** [comment]
Current issues:

For macOS build: 
`md5sum: command not found`
 - I don't really understand why [this error](https://github.com/mlcommons/cm4mlops/actions/runs/11626277757/job/32377636901#step:6:10237) occurs except that installation wasn't successful With the correct software, it should work. Tested on a macbook.
 
For ubuntu:
dpkg -l | grep libnuma-dev
no packages found matching libnuma-dev

Not sure why this errors... on a numa supported device it shld work. Tested to be working on [docker from runner](https://github.com/actions/runner/pkgs/container/actions-runner/283808981?tag=2.320.0) and ubuntu bare metal.

**Comment by arjunsuresh** [comment]
@Oseltamivir I guess I have fixed the errors on macos - the problem is `md5sum --version` works fine but it still returns `255` on macos. So, added a `grep` suffix to the version detect command. Also, for the newly added version regular expression the group number is 0 and not 1.

File: script/get-generic-sys-util/_cm.json [hunk 1]
```diff
@@ -80,11 +82,14 @@
...
-      }
+      },
+      "new_env_keys": ["CM_CMAKE_VERSION"]
     },
     "dmidecode": {
       "env": {
-        "CM_SYS_UTIL_NAME": "dmidecode"
+        "CM_SYS_UTIL_NAME": "dmidecode",
+        "CM_SYS_UTIL_VERSION_CMD": "dmidecode --version",
+        "CM_SYS_UTIL_VERSION_RE": "([\\d.]+)"
```
[the remark this answers is not in the document]
Comment (Oseltamivir) [reply script/get-generic-sys-util/_cm.json:92]: More of these shld be added, I'll add this to all of them

File: script/get-generic-sys-util/_cm.json [hunk 2]
```diff
@@ -525,19 +631,24 @@
         }
       ],
       "env": {
-        "CM_SYS_UTIL_NAME": "numactl"
+        "CM_SYS_UTIL_NAME": "numactl",
+        "CM_SYS_UTIL_VERSION_CMD": "dpkg -l | grep libnuma",
+        "CM_SYS_UTIL_VERSION_RE": "\\b\\d+\\.\\d+(?:\\.\\d+)?\\b"
```
[the remark this answers is not in the document]
Comment (Oseltamivir) [reply script/get-generic-sys-util/_cm.json:636]: same as above, I'll add with others

**Comment by Oseltamivir** [comment]
@arjunsuresh I'm gonna work on the CM_REPO_PATH part for a bit, I can only access an ubuntu server on weekdays to properly test the scripts. Running docker on my macbook right now is quite limiting. 

Is next week's meet still using the link: https://meet.google.com/jtf-crbz-ezz?  ?

Also, if it is possible, I hope my contributions can count towards my team's points in SCC24

**Comment by arjunsuresh** [comment]
Hi @Oseltamivir I removed the version detection ones which are failing and those which reply on `dpkg` or `pkg_config` as currently we assume that when version detection of a package is successful, the package is installed and available. We can fix these later. Currently all the tests are passing.

File: automation/script/module.py [hunk 3]
```diff
@@ -3979,7 +3979,12 @@ def parse_version(self, i):
            return r
 
         string = r['string']
-
+        match = re.search(match_text, string)
```
Comment (arjunsuresh) [inline automation/script/module.py:3982]: We don't need this right?

**Review by arjunsuresh — APPROVED** [review]
Thank you for the changes

**Pull request merged**